### PR TITLE
[WIP] Nov15 - Make push only mandatory when fork activates

### PR DESCRIPTION
**Pull request by ptschip**
Will wait for the activation code to be in place as well as the activation py tests before finalizing this PR>

### CODE REVIEW

File: qa/rpc-tests/nov152018_forkactivation.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+
+    def set_test_params(self):
+        self.num_nodes = 1
+        self.setup_clean_chain = True
+        self.block_heights = {}
+        self.tip = None
+        self.blocks = {}
+        self.extra_args = [['-whitelist=127.0.0.1',
+                            "-magneticanomalyactivationtime=%d" % MAGNETIC_ANOMALY_START_TIME,
+                            "-replayprotectionactivationtime=%d" % (2 * MAGNETIC_ANOMALY_START_TIME)]]
```
Comment (sickpig) [inline qa/rpc-tests/nov152018_forkactivation.py:40]: shouldn't we need to remove at least `replayprotectionactivationtime`, we decided to no implement the automatic reply protection (i.e. fork time bomb)

File: qa/rpc-tests/nov152018_forkactivation.py [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+from test_framework.util import assert_equal, assert_raises_rpc_error
+from test_framework.comptool import TestManager, TestInstance, RejectResult
+from test_framework.blocktools import create_coinbase, create_block
+from test_framework.mininode import *
+from test_framework.script import *
+from test_framework.cdefs import MIN_TX_SIZE
+from collections import deque
+
+# far into the future
+MAGNETIC_ANOMALY_START_TIME = 2000000000
```
Comment (sickpig) [inline qa/rpc-tests/nov152018_forkactivation.py:20]: I would rather call it `NOV152018_START_TIME`

File: qa/rpc-tests/nov152018_forkactivation.py [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+
+
+class PreviousSpendableOutput():
+
+    def __init__(self, tx=CTransaction(), n=-1):
+        self.tx = tx
+        self.n = n  # the output we're spending
+
+
+class MagneticAnomalyActivationTest(ComparisonTestFramework):
```
Comment (sickpig) [inline qa/rpc-tests/nov152018_forkactivation.py:30]: I would also change the name of this `class` since we are not using ABC nomenclature.

**Review by sickpig — COMMENTED** [review]
I know this is WIP but since I was looking at the code I decide to write down my feedback before forgetting about it :smile:

**Comment by ptschip** [comment]
don't worry about the py tests for now, I just included the file for
general reference, but i think we'll probably end up possibly cutting
and pasting these into may152018_forkactivation.py tests (if it even
makes sense) since those are our templates for fork activation.

**Comment by ptschip** [comment]
closing in favor of #1376

**Pull request closed without merging**